docs/middleware/loadshedding.md

```diff
@@ -0,0 +1,81 @@
+---
+id: loadshedding
+---
+
+# Load Shedding
+
+The **Load Shedding** middleware for [Fiber](https://github.com/gofiber/fiber) helps maintain server stability by applying request-processing timeouts. It prevents resource exhaustion by gracefully rejecting requests that exceed a specified time limit. This is particularly beneficial in high-traffic scenarios, where preventing overload is crucial to sustaining service availability and performance.
+
+## Features
+
+- **Request Timeout Enforcement**: Automatically terminates any request that exceeds the configured processing time.
+- **Customizable Response**: Enables you to define a specialized response for timed-out requests.
+- **Exclusion Logic**: Lets you skip load-shedding for specific requests, such as health checks or other critical endpoints.
+- **Enhanced Stability**: Helps avoid server crashes or sluggish performance under heavy load by shedding excess requests.
+
+## Use Cases
+
+- **High-Traffic Scenarios**: Safeguard critical resources by rejecting overly long or resource-intensive requests.
+- **Health Check Protection**: Exclude monitoring endpoints (e.g., `/health`) to ensure uninterrupted external checks.
+- **Dynamic Load Management**: Utilize exclusion logic to adjust load-shedding behavior for specific routes or request types.
+
+---
+
+## Signature
+
+```go
+func New(timeout time.Duration, loadSheddingHandler fiber.Handler, exclude func(fiber.Ctx) bool) fiber.Handler
```

Comment:
in order to ensure expandability in the future and not have to make breaking changes, i would like to stick with the configuration concept, which we use in all other middlewares and which has proven to be very easy to expand